include/BALL/KERNEL/bond.iC

```diff
@@ -32,12 +32,20 @@ void Bond::destroy()
 BALL_INLINE
 void Bond::finalize()
 {
-	if (*first_ > *second_)
-	{
-		Atom* tmp = first_;
-		first_ = second_;
-		second_ = tmp;
-	}
+        if (first_ && second_)
+        {
+                if (*first_ > *second_)
+                {
+                        Atom* tmp = first_;
+                        first_ = second_;
+                        second_ = tmp;
+                }
+        }
+        else
```

Comment:
Yes, I actually believe the test is wrong. It serialises a default constructed Bond to a file and then deserialises it. It can be fixed simply by also creating two dummy Atoms that are added to the Bond. I'll push two patches that fix the test and add an assertion to Bond::finalize().